src/core/Menu/Menu.tsx

```diff
@@ -35,49 +47,34 @@ export const Menu = React.forwardRef<HTMLUListElement, MenuProps>(
     const refs = useMergedRefs(menuRef, ref);
 
     React.useEffect(() => {
-      const items = menuRef.current?.children;
+      const items = getFocusableItems(menuRef.current);
       if (focusedIndex != null) {
         (items?.[focusedIndex] as HTMLLIElement)?.focus();
         return;
       }
 
-      const childrenArray = React.Children.toArray(children);
-      const selectedIndex = childrenArray.findIndex(
-        (child: JSX.Element) => child.props.isSelected,
-      );
-      const firstEnabledIndex = childrenArray.findIndex(
-        (child: JSX.Element) => !child.props.disabled,
+      const selectedIndex = items.findIndex((el) =>
+        el.classList.contains('iui-active'),
```

Comment:
I am not sure 😀 Either way if prop changes or class changes, additional change will need to be done here too and our tests should catch it. Just handling keyboard navigation using HTML elements looked much simpler and it is easier to handle profile menu case.

Reply:
but it's additional change in same repo vs different repo.

Also we talk about "custom component", so maybe we could target something agnostic like `aria-selected`.

Reply:
I really your suggestion - `aria-selected`! It would help with custom items too because developers would just need to add `aria-selected` whether item is selected or not.

Reply:
Checking `aria-selected` instead. I couldn't write `el.ariaSelected === 'true'` because for some reason it is `undefined` in tests.

Reply:
>  I couldn't write `el.ariaSelected === 'true'`
I thought it's a boolean not a string?

And does it work outside of tests?

Reply:
Yeah.